modules/EvManager/main/car_simulatorImpl.cpp

```diff
@@ -0,0 +1,586 @@
+        }
+        simulator->simData->state = SimState::CHARGING_FIXED;
+        return true;
+    };
+    commandRegistry->registerCommand("draw_power_fixed", 2, draw_power_fixedFunction);
+
+    const auto pauseFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::PLUGGED_IN;
+        return true;
+    };
+    commandRegistry->registerCommand("pause", 0, pauseFunction);
+
+    const auto unplugFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::UNPLUGGED;
+        return true;
+    };
+    commandRegistry->registerCommand("unplug", 0, unplugFunction);
+
+    const auto error_eFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::ERROR_E;
+        return true;
+    };
+    commandRegistry->registerCommand("error_e", 0, error_eFunction);
+
+    const auto diode_failFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::DIODE_FAIL;
+        return true;
+    };
+    commandRegistry->registerCommand("diode_fail", 0, diode_failFunction);
+
+    const auto rcd_currentFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->rcd_current_mA = std::stof(arguments[0]);
+        return true;
+    };
+    commandRegistry->registerCommand("rcd_current", 1, rcd_currentFunction);
+
+    const auto iso_wait_slac_matchedFunction = [](car_simulatorImpl* const simulator,
+                                                  const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::PLUGGED_IN;
+
+        if (simulator->mod->r_slac.empty()) {
+            EVLOG_debug << "Slac undefined";
+        }
+        if (simulator->simData->slacState == "UNMATCHED") {
+            EVLOG_debug << "Slac UNMATCHED";
+            if (!simulator->mod->r_slac.empty()) {
+                EVLOG_debug << "Slac trigger matching";
+                simulator->mod->r_slac[0]->call_reset();
+                simulator->mod->r_slac[0]->call_trigger_matching();
+                simulator->simData->slacState = "TRIGGERED";
+            }
+        }
+        if (simulator->simData->slacState == "MATCHED") {
+            EVLOG_debug << "Slac Matched";
+            return true;
+        }
+        return false;
+    };
+    commandRegistry->registerCommand("iso_wait_slac_matched", 0, iso_wait_slac_matchedFunction);
+
+    if (!mod->r_ev.empty()) {
+        const auto iso_start_v2g_sessionFunction = [](car_simulatorImpl* const simulator,
+                                                      const std::vector<std::string>& arguments) {
+            const auto& argument = arguments[0];
+            if (argument == "ac_single_phase_core") {
```

Comment:
Define enum. E.g. `class enum EnergyMode : std::string`

Reply:
Makes sense. I had to create this myself though